azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/features/VirtualMachineApiLiveTest.java

```diff
@@ -0,0 +1,269 @@
+import org.jclouds.azurecompute.arm.domain.ImageReference;
+import org.jclouds.azurecompute.arm.domain.NetworkInterfaceCard;
+import org.jclouds.azurecompute.arm.domain.NetworkProfile;
+import org.jclouds.azurecompute.arm.domain.OSDisk;
+import org.jclouds.azurecompute.arm.domain.OSProfile;
+import org.jclouds.azurecompute.arm.domain.StorageProfile;
+import org.jclouds.azurecompute.arm.domain.StorageService;
+import org.jclouds.azurecompute.arm.domain.VHD;
+import org.jclouds.azurecompute.arm.domain.VirtualMachine;
+import org.jclouds.azurecompute.arm.domain.VirtualMachineInstance;
+import org.jclouds.azurecompute.arm.domain.VirtualMachineProperties;
+import org.jclouds.azurecompute.arm.functions.ParseJobStatus;
+import org.jclouds.azurecompute.arm.internal.BaseAzureComputeApiLiveTest;
+import org.jclouds.util.Predicates2;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.testng.Assert.assertTrue;
+
+@Test(groups = "live", testName = "VirtualMachineApiLiveTest")
+public class VirtualMachineApiLiveTest extends BaseAzureComputeApiLiveTest {
+
+   private String subscriptionid = getSubscriptionId();
+   private String vmName = null;
+   private String nicName = null;
+
+   @BeforeClass
+   public void Setup() {
+      NetworkInterfaceCard nic = getOrCreateNetworkInterfaceCard(NETWORKINTERFACECARD_NAME);
+      assertNotNull(nic);
+      nicName = nic.name();
+   }
+
+   private String getName() {
+      if (vmName == null) {
+         vmName = String.format("%3.24s",
+                 System.getProperty("user.name") + RAND + this.getClass().getSimpleName()).toLowerCase().substring(0, 15);
+      }
+      return vmName;
+   }
+
+   @Test
+   public void testCreate() {
+      StorageAccountApi storageApi = api.getStorageAccountApi(getResourceGroupName());
+      StorageService storageAccount = storageApi.get(getStorageServiceName());
+      String blob = storageAccount.storageServiceProperties().primaryEndpoints().get("blob");
+
+      String id = "/subscriptions/" + subscriptionid + "/resourceGroups/" + getResourceGroupName() +
+              "/providers/" + "Microsoft.Compute/virtualMachines/" + getName();
+      VirtualMachine vm = api().create(getName(), id, getName(), LOCATION, getProperties(blob, nicName));
+      assertTrue(!vm.name().isEmpty());
+      String status = "Creating";
+      int stopCounter = 0;
+      while (status.equals("Creating")) {
+         try {
+            Thread.sleep(120 * 1000); // Creation takes long time. Recheck later
+         } catch (InterruptedException e) {
+            e.printStackTrace();
+         }
+         vm = api().get(getName());
+         status = vm.properties().provisioningState();
+         stopCounter++;
+         if (stopCounter == 15)
+            break;
+      }
```

Comment:
Same as before, use the jclouds `Predictes2.retry` construct, and if it makes sense create a predicate and make it injectable for later reuse. The `AzureComputeHttpApiModule` is a good place to declare such reusable predicates.